divide_and_conquer/binary_exponentiation.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file
+ * @brief Implementation of Binary Exponentiation
+ * @details
+ * Binary Exponentiation is a way to calculate exponential
+ * calculations using much less time complexity (O(logn))
+ * using a divide and conquer technique through recursion.
+ * [https://cp-algorithms.com/algebra/binary-exp.html]
+ * @author [Timothy Chiles](https://github.com/TSChiles)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+
+/**
+ * @namespace divide_and_conquer
+ */
+namespace divide_and_conquer {
+
+/**
+ * @brief Function documentation
+ * @param param1 is the base number
+ * @param param2 is the exponent
+ * @returns the base to the exponent operation
+ */
+    long long expo(long long param1, long long param2) {
+        if (!param2){
+            return 1;
+        }
+        //recursion until reaching 0
+        long long sol = expo(param1, param2/2);
+        //if odd, multiply by base again
+        if(param2%2){
```

Comment:
In both cases you return sol². I suggest you create a variable that stores this and follow the flow:

Example :
```
    long long sun = expo(param1, param2/2);
    //if odd, multiply by base again
    long long squaredSol = sun * sun;
    if(param2%2){
        return squaredSol * param1;
    }
   return squaredSol ;

```
just one suggestion  #goodpratices :)